nucypher/cli/stake.py

```diff
@@ -0,0 +1,310 @@
+
+import click
+from constant_sorrow.constants import NO_STAKING_DEVICE
+from web3 import Web3
+
+from nucypher.blockchain.eth.actors import StakeHolder
+from nucypher.blockchain.eth.interfaces import BlockchainInterface
+from nucypher.blockchain.eth.registry import EthereumContractRegistry
+from nucypher.blockchain.eth.token import NU
+from nucypher.characters.banners import NU_BANNER
+from nucypher.cli import painting
+from nucypher.cli.actions import confirm_staged_stake, get_password, select_stake, select_client_account
+from nucypher.cli.config import nucypher_click_config
+from nucypher.cli.types import (
+    EIP55_CHECKSUM_ADDRESS,
+    STAKE_VALUE,
+    STAKE_DURATION,
+    STAKE_EXTENSION,
+    EXISTING_READABLE_FILE)
+
+
+@click.command()
+@click.argument('action')
+@click.option('--config-root', help="Custom configuration directory", type=click.Path())
+@click.option('--config-file', help="Path to configuration file", type=EXISTING_READABLE_FILE)
+@click.option('--force', help="Don't ask for confirmation", is_flag=True)
+@click.option('--quiet', '-Q', help="Disable logging", is_flag=True)
+@click.option('--device/-no-device', help="Use a hardware wallet for staking operations", is_flag=True, default=False)  # TODO: Make True or deprecate.
+@click.option('--registry-filepath', help="Custom contract registry filepath", type=EXISTING_READABLE_FILE)
+@click.option('--poa', help="Inject POA middleware", is_flag=True)
```

Comment:
I guess, one gotcha: format of the provider uri is different from deployment: it's `file:///tmp/geth.ipc` rather than `ipc:///tmp/geth.ipc`. Should we make our mind, which one to use?

Reply:
Oh I dont intend for them to be different, although I think we support both for ease.  I prefer `file://` because IPC can mean other forms of comm.

Reply:
Totally fine if we support both!

Reply:
:heavy_check_mark: